src/Features/Core/Portable/GoToDefinition/AbstractFindDefinitionService.cs

```diff
@@ -18,9 +20,11 @@ internal class AbstractFindDefinitionService : IFindDefinitionService
             var symbolService = document.GetRequiredLanguageService<IGoToDefinitionSymbolService>();
             var (symbol, _) = await symbolService.GetSymbolAndBoundSpanAsync(document, position, includeType: true, cancellationToken).ConfigureAwait(false);
 
+            symbol = await SymbolFinder.FindSourceDefinitionAsync(symbol, document.Project.Solution, cancellationToken).ConfigureAwait(false) ?? symbol;
```

Comment:
It appears that LSP didn't handle cross-language go-to-def properly either; I suspect this is fixing it.

Reply:
Well done and thank you! I'm crediting this change with fixing https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1433741 😁😁😁